### ISSUE TITLE
Where did semiauto.deriveUnwrappedValueMapper go?

### ISSUE THREAD

**Issue by PerWiklander** [neotypes/neotypes#815]
I'm late in updating my dependencies. I was using `semiauto.deriveUnwrappedValueMapper` for my AnyVals. Is there an alternative today?

**Comment by BalmungSan** [comment]
No there is no direct alternative.
**Scala 3** soft deprecated `AnyVal`, and I was really in favor of encouraging `AnyVal` in **Scala 2** since I am convinced most people used them wrong. In any case, AFAIK, **Scala 3** meta-programming don't provide any way to detect `AnyVals` so, again AFAIK _(please correct me if wrong)_, many libraries like **circe** are not longer providing special treatment to those anymore as well.

As of today, you can create a `ResultMapper` very easily using: `wrappedType.map(WrapperClass.apply)`
In my TODO list, I have pending adding a combinator to simplify that a bit, but still not a big deal.

**Comment by PerWiklander** [comment]
Thanks, that was what I was looking for. This is for my Scala 2 code base. In Scala 3 I will use normal opaque types (with the help of https://github.com/kitlangton/neotype).

**Comment by BalmungSan** [comment]
@PerWiklander yeah so for consistency with the API since we don't have an equivalent for **Scala 3** I removed it for **Scala 2**.
You may take a look at the source code how it was before: https://github.com/neotypes/neotypes/blob/v0.23.3/generic/src/main/scala/neotypes/generic/semiauto.scala#L18-L21 and re-add it to your projects.

**Comment by PerWiklander** [comment]
Btw, will opaque types just work? If they are subtypes of their wrapped types.

**Comment by BalmungSan** [comment]
@PerWiklander uhm it depends on your definition of just work.
You can't use a `Decoder[Int]` to get back a `PositiveInt`, but creating a `Decoder[PositiveInt]` should be as simple as `Decoder.int.emap(PositiveInt.apply)` assuming `PositiveInt.apply` returns a `Either[Error, PositiveInt]`.

**Comment by PerWiklander** [comment]
Lets say the returned data does not validate, So that the `Either` is a `Left`. Would that result in a failed future from the query, just like if e.g. I have a typo in the cypher so that the field is missing?

**Comment by BalmungSan** [comment]
> Would that result in a failed future from the query

Yes, it would.

BTW, if the `opaque` does not validate but is just a simple wrapper you can also just use `map(PositiveInt.apply)`.
